### PR TITLE
Added function-constructor rule

### LINKED ISSUE #3554: Port tslint-microsoft-contrib's no-function-constructor-with-string-args
See https://github.com/palantir/tslint/issues/1142.

> `no-function-constructor-with-string-args` can now done as an optionally typed rule: without type info it only flags string literals. It should probably also include `setInterval` and `setTimeout`, with a config to add other built-in functions that allow strings as functions.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### PR checklist

- [x] Addresses an existing issue: #3554
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:

Blocks using the `Function()`/`new Function()` methods to create new functions.

Fixes #3554.

#### CHANGELOG.md entry:

[new-rule] `function-constructor`

### CODE REVIEW

File: src/configs/all.ts [hunk 1]
```diff
@@ -29,19 +29,25 @@ export const rules = {
     "ban-types": {
```
Comment (JoshuaKGoldberg) [inline src/configs/all.ts]: Err. I think my prettier plugin did this automatically. Will revert...

**Comment by ericanderson** [comment]
@JoshuaKGoldberg Since I'm new to maintaining this, I want to make sure we are doing the right things. Doesn't changing "recommended" constitute a break?

**Comment by JoshuaKGoldberg** [comment]
Heya @ericanderson - technically yes, but historically it's had additive changes in patch versions. For example, `5.11.0` added the `no-reference-import` rule to recommended. IMO this is fine because the `recommended`/`all` lists don't come with a guarantee that nothing will be added, as with TypeScript's `--strict` flag. It'd also be pretty restrictive if those rulesets could only be changed in major versions.

If this were removing a rule from `recommend` or breaking behavior of an existing rule, I'd feel differently.

Does that sit right with you? I can remove the changed to `recommended.ts` if you want. 😊

**Comment by adidahiya** [comment]
@JoshuaKGoldberg @ericanderson actually, we shouldn't be making `tslint:recommended` more strict across minor versions, only very obvious bug fixes should cause a change to that rule set. This is documented under "Configuration presets" here: https://palantir.github.io/tslint/usage/configuration/

> This configuration follows semver, so it will not have breaking changes across minor or patch releases.

`tslint:latest` is the rule set which does not follow semver:

> Using this config may introduce breaking changes across minor releases as new rules are enabled which cause lint failures in your code.

**Comment by JoshuaKGoldberg** [comment]
Darn, thanks for the clarification @adidahiya!

**Comment by ericanderson** [comment]
I wanted to make it less strict :)

**Comment by adidahiya** [comment]
@ericanderson I agree with that overall goal in v6.0, through #4305

**Pull request merged**